src/main/java/com/patchworkmc/event/EventHandlerRewriter.java

```diff
@@ -66,35 +83,84 @@ public MethodVisitor visitMethod(int access, String name, String descriptor, Str
 		return new MethodScanner(parent, access, name, descriptor, signature);
 	}
 
+	/**
+	 * @deprecated glitch please clean this code up before merging it
```

Comment:
This needs some work, splitting out a method for instance handlers and one for static handlers would be a good start (in line with "Just have two lists, one for the annotations on instance methods and one for the annotations on static methods"). This would probably help avoid the need for superfluous null checks, too.

Also, the lines in this method are *very* long all around. It'll be easier to make more comments after some initial cleanup.